app/src/main/java/com/hover/stax/database/DatabaseRepo.kt

```diff
@@ -169,7 +176,13 @@ class DatabaseRepo(db: AppDatabase, sdkDb: HoverRoomDatabase) {
         }
     }
 
-
+    private fun shouldFlagFailure(transaction: StaxTransaction, c: Context?) : Boolean {
```

Comment:
This name is confusing, should be something like `isSessionIncomplete` and it should be a method on the transaction model. You also don't need to use the UssdCallResponse object or creation method at all, you can just use `Hover.getTransaction(transaction.uuid, c).ussdMessages`

Reply:
Thanks, changes made